logstash-core-event-java/src/main/java/com/logstash/Valuefier.java

```diff
@@ -0,0 +1,105 @@
+package com.logstash;
```

Comment:
The `Javafier` only deals with Ruby objects. I chose not to make it aware of `RubyJavaValue` because it is a Java object.

The `Rubyfier` only deals with Java objects and unwraps `RubyJavaValue` values.

The `Valuefier` deals with either Ruby or Java objects.
It converts them to their opposite equivalents and wraps values in a `RubyJavaValue`.
It unconverts on the Java side and unwraps RubyJavaValue values to Java.

`Javafier.deep` is now only called in `Valuefier` or `RubyJavaValue`
`Rubyfier.deep` is only called in `RubyJavaValue` or the JRuby Extension.

BTW, there is a bug in the `unconvertMap` method - calls convert instead of unconvert. (will fix)